default.go

```diff
@@ -109,7 +113,20 @@ func IsHostname(str string) bool {
 			valid = false
 		}
 	}
-	return valid
+
+	if !valid {
+		return false
+	}
```

Comment:
See https://github.com/go-openapi/strfmt/pull/150#discussion_r1996304481 about why I suggest removing this

```suggestion
```